sdk/batch/batch-rest/generated/parameters.ts

```diff
@@ -0,0 +1,3526 @@
+  "return-client-request-id"?: boolean;
+  /**
+   * The time the request was issued. Client libraries typically set this to the
+   * current system clock time; set it explicitly if you are calling the REST API
+   * directly.
+   */
+  "ocp-date"?: string;
+}
+
+export interface ListPoolsQueryParamProperties {
+  /** The maximum time that the server can spend processing the request, in seconds. The default is 30 seconds. If the value is larger than 30, the default will be used instead.". */
+  timeOut?: number;
+  /**
+   * The maximum number of items to return in the response. A maximum of 1000
+   * applications can be returned.
+   */
+  maxresults?: number;
+  /**
+   * An OData $filter clause. For more information on constructing this filter, see
+   * https://docs.microsoft.com/en-us/rest/api/batchservice/odata-filters-in-batch#list-pools.
+   */
+  $filter?: string;
+  /** An OData $select clause. */
+  $select?: string[];
+  /** An OData $expand clause. */
+  $expand?: string[];
+}
+
+export interface ListPoolsQueryParam {
+  queryParameters?: ListPoolsQueryParamProperties;
+}
+
+export interface ListPoolsHeaderParam {
+  headers?: RawHttpHeadersInput & ListPoolsHeaders;
+}
+
+export type ListPoolsParameters = ListPoolsQueryParam &
+  ListPoolsHeaderParam &
+  RequestParameters;
+
+export interface DeletePoolHeaders {
+  /**
+   * The caller-generated request identity, in the form of a GUID with no decoration
+   * such as curly braces, e.g. 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0.
+   */
+  "client-request-id"?: string;
+  /** Whether the server should return the client-request-id in the response. */
+  "return-client-request-id"?: boolean;
+  /**
+   * The time the request was issued. Client libraries typically set this to the
+   * current system clock time; set it explicitly if you are calling the REST API
+   * directly.
+   */
+  "ocp-date"?: string;
+  /**
+   * A timestamp indicating the last modified time of the resource known to the
+   * client. The operation will be performed only if the resource on the service has
+   * been modified since the specified time.
+   */
+  "If-Modified-Since"?: string;
+  /**
+   * A timestamp indicating the last modified time of the resource known to the
+   * client. The operation will be performed only if the resource on the service has
+   * not been modified since the specified time.
+   */
+  "If-Unmodified-Since"?: string;
+  /**
+   * An ETag value associated with the version of the resource known to the client.
+   * The operation will be performed only if the resource's current ETag on the
+   * service exactly matches the value specified by the client.
+   */
+  "If-Match"?: string;
+  /**
+   * An ETag value associated with the version of the resource known to the client.
+   * The operation will be performed only if the resource's current ETag on the
+   * service does not match the value specified by the client.
+   */
+  "If-None-Match"?: string;
```

Comment:
There eixsts gaps between guidelines and RLC, not sure we should require customizations in
https://azure.github.io/azure-sdk/typescript_design.html#ts-conditional-requests.

Reply:
@MaryGao would it be possible for code gen to generate those options? Then we could mark these as deprecated and remove them in later version.